internal/controllers/clusterextension_controller.go

```diff
@@ -193,16 +193,17 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	l.V(1).Info("handling finalizers")
 	finalizeResult, err := r.Finalizers.Finalize(ctx, ext)
 	if err != nil {
-		// TODO: For now, this error handling follows the pattern of other error handling.
-		//  Namely: zero just about everything out, throw our hands up, and return an error.
-		//  This is not ideal, and we should consider a more nuanced approach that resolves
-		//  as much status as possible before returning, or at least keeps previous state if
-		//  it is properly labeled with its observed generation.
-		setInstallStatus(ext, nil)
 		setResolutionStatus(ext, nil)
 		setResolvedStatusConditionFailed(ext, err.Error())
 		setStatusProgressing(ext, err)
-		ensureAllConditionsWithReason(ext, ocv1alpha1.ReasonFailed, err.Error())
+
+		// Here we check if the bundle is already installed.
+		// We only set Installed status to False if the bundle is actually no longer on cluster.
+		installedBundle, _ := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
+		if installedBundle == nil {
+			setInstallStatus(ext, nil)
+			setInstalledStatusConditionFailed(ext, err.Error())
```

Comment:
I know we talked about this about during our sync yesterday, but wanted to pose the question here as well in case other maintainers have opinions on it:

Do we need to set the Installed condition to False here and overwrite any errors that may exist in it with the finalizer error?

My feeling is that we do not need to do this as a user may want to still be able to see the original failed installation message. We already set the Progressing condition with the finalizer error so I think that should be sufficient here.

Additionally, if the Installed condition did not previously exist, I don't think there is any reason to set it here. Non-existent Installed condition should be interpreted as "Unknown".

Maybe a more fundamental question (that popped into my head as I was typing ^) - if there is no reason for us to change any of the status related to installation of the extension can we just remove any logic to do with the installed status from the finalizer error handling?